presto-spi/src/main/java/com/facebook/presto/spi/SplitContext.java

```diff
@@ -13,25 +13,42 @@
  */
 package com.facebook.presto.spi;
 
+import com.facebook.presto.common.predicate.TupleDomain;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
+import java.util.Optional;
+
 // TODO: Use builder pattern for SplitContext if we are to add optional field
 public class SplitContext
 {
     public static final SplitContext NON_CACHEABLE = new SplitContext(false);
 
     private final boolean cacheable;
+    // This is not serializable; for local execution only
```

Comment:
the data structure is serializable. It's "// For local execution only; no need for serialization."